google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/ErrorCountPerConnectionTest.java

```diff
@@ -0,0 +1,261 @@
+import io.grpc.*;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+@RunWith(JUnit4.class)
+public class ErrorCountPerConnectionTest {
+  private static final String SUCCESS_TABLE_NAME = "some-table";
+  private static final String ERROR_TABLE_NAME = "nonexistent-table";
+  private Server server;
+  private final FakeService fakeService = new FakeService();
+  private EnhancedBigtableStubSettings.Builder builder;
+  private ArgumentCaptor<Runnable> runnableCaptor;
+  private StatsRecorderWrapper statsRecorderWrapper;
+
+  @Before
+  public void setup() throws Exception {
+    server = FakeServiceBuilder.create(fakeService).start();
+
+    ScheduledExecutorService executors = Mockito.mock(ScheduledExecutorService.class);
+    builder =
+        BigtableDataSettings.newBuilderForEmulator(server.getPort())
+            .stubSettings()
+            .setBackgroundExecutorProvider(FixedExecutorProvider.create(executors))
+            .setProjectId("fake-project")
+            .setInstanceId("fake-instance");
+    runnableCaptor = ArgumentCaptor.forClass(Runnable.class);
+    System.out.println("rez1 " + executors + " 2 = " + builder);
+    Mockito.when(
+            executors.scheduleAtFixedRate(runnableCaptor.capture(), anyLong(), anyLong(), any()))
+        .thenReturn(null);
+
+    statsRecorderWrapper = Mockito.mock(StatsRecorderWrapper.class);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (server != null) {
+      server.shutdown();
+    }
+  }
+
+  @Test
+  public void singleRead() throws Exception {
+    EnhancedBigtableStub stub = EnhancedBigtableStub.create(builder.build());
+    long errorCount = 0;
+
+    for (int i = 0; i < 20; i++) {
+      Query query;
+      if (i % 3 == 0) {
+        query = Query.create(ERROR_TABLE_NAME);
+        errorCount += 1;
+      } else {
+        query = Query.create(SUCCESS_TABLE_NAME);
+      }
+      try {
+        stub.readRowsCallable().call(query).iterator().hasNext();
+      } catch (Exception e) {
+        // noop
+      }
+    }
+    ArgumentCaptor<Long> errorCountCaptor = ArgumentCaptor.forClass(long.class);
+    Mockito.doNothing()
+        .when(statsRecorderWrapper)
+        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
+    runInterceptorTasksAndAssertCount(1);
+    List<Long> allErrorCounts = errorCountCaptor.getAllValues();
+    assertThat(allErrorCounts.size()).isEqualTo(1);
+    assertThat(allErrorCounts.get(0)).isEqualTo(errorCount);
+  }
+
+  @Test
+  public void readWithTwoStubs() throws Exception {
+    EnhancedBigtableStub stub1 = EnhancedBigtableStub.create(builder.build());
```

Comment:
Instead of creating 2 stubs, you can set the number of channels in EnhancedBigtableStubSettings. For example: `stubSettingsBuilder.setTransportChannelProvider(((InstantiatingGrpcChannelProvider) stubSettingsBuilder.getTransportChannelProvider()).toBuilder().setChannelPoolSettings(ChannelPoolSettings.staticallySized(2)).build());` this gives you 2 channels.